apis/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -81,6 +81,12 @@ type OpenTelemetryCollectorSpec struct {
 	// +optional
 	// +listType=atomic
 	Ports []v1.ServicePort `json:"ports,omitempty"`
+	// ContainerPorts allows a set of ports to be exposed by the underlying v1.Container. By default, the operator
+	// will attempt to infer the required ports by parsing the .Spec.Config property but this property can be
+	// used to open additional ports that can't be inferred by the operator, like for custom receivers.
```

Comment:
I believe the container ports in the spec will also override any inferred names, maybe we should add that in to this comment